Extensions/PathfindingBehavior/tests/legacypathfindingruntimebehavior.spec.js

```diff
@@ -0,0 +1,689 @@
+      playerLeftBorder = 80;
+      playerRightBorder = 80;
+      playerTopBorder = 80;
+      playerBottomBorder = 80;
+    } else {
+      playerLeftBorder = 87;
+      playerRightBorder = 73;
+      playerTopBorder = 87;
+      playerBottomBorder = 73;
+    }
+  } else {
+    // The player radius is at most 2 cells
+    playerLeftBorder = 2 * cellSize;
+    playerRightBorder = 2 * cellSize;
+    playerTopBorder = 2 * cellSize;
+    playerBottomBorder = 2 * cellSize;
+  }
+
+  if (objectCenteredOnCells) {
+    obstacleLeftBorder = 80;
+    obstacleRightBorder = 80;
+    obstacleTopBorder = 80;
+    obstacleBottomBorder = 80;
+  } else {
+    obstacleLeftBorder = 87;
+    obstacleRightBorder = 73;
+    obstacleTopBorder = 87;
+    obstacleBottomBorder = 73;
+  }
+
+  let inclusiveMargin;
+  if (cellSize == 160) {
+    // The collision is strict
+    inclusiveMargin = 0;
+  } else {
+    // The collision is not strict
+    inclusiveMargin = 1;
+  }
+
+  // multiple of 20 and 160
+  const playerX = 480;
+  const playerY = 320;
+
+  // This one is not on limit
+  it('can find a path with an obstacle near the start', function () {
+    let obstacleX;
+    let obstacleY;
+    let targetX;
+    let targetY;
+    switch (direction) {
+      case 'right':
+        // right on the player right
+        obstacleX = playerX + playerRightBorder + obstacleLeftBorder;
+        obstacleY = playerY;
+        targetX = playerX - 2 * cellSize;
+        targetY = playerY;
+        break;
+      case 'left':
+        obstacleX = playerX - playerLeftBorder - obstacleRightBorder;
+        obstacleY = playerY;
+        targetX = playerX + 2 * cellSize;
+        targetY = playerY;
+        break;
+      case 'up':
+        obstacleX = playerX;
+        obstacleY = playerY - playerTopBorder - obstacleBottomBorder;
+        targetX = playerX;
+        targetY = playerY + 2 * cellSize;
+        break;
+      case 'down':
+        obstacleX = playerX;
+        obstacleY = playerY + playerBottomBorder + obstacleTopBorder;
+        targetX = playerX;
+        targetY = playerY - 2 * cellSize;
+        break;
+    }
+
+    obstacle.setPosition(obstacleX, obstacleY);
+    // so obstacles register
+    runtimeScene.renderAndStep(1000 / 60);
+
+    // move away from the obstacle
+    player.setPosition(playerX, playerY);
+    player.getBehavior(pathFindingName).moveTo(runtimeScene, targetX, targetY);
+    //expect(player._animationFrame).to.be(null);
+    expect(player.getBehavior(pathFindingName).pathFound()).to.be(true);
+    expect(player.getBehavior(pathFindingName).getNodeCount()).to.be(3);
+  });
+
+  it('can find a path when an obstacle is overlaping only at the start', function () {
+    let obstacleX;
+    let obstacleY;
+    let targetX;
+    let targetY;
+    switch (direction) {
+      case 'right':
+        // right on the player right
+        obstacleX =
+          playerX +
+          playerRightBorder +
+          obstacleLeftBorder -
+          cellSize +
```

Comment:
Why removing a cellSize in all of these coordinates?

Reply:
This needs a comment and a better test title. It's testing that if an obstacle is place so it will be right next to the player after the first step the player is not stuck.